src/main/webapp/app/markdown-editor/markdown-editor.component.ts

```diff
@@ -0,0 +1,276 @@
+             * )  close the bracket
+             *  g: search in the whole string
+             *  m: match the regex over multiple lines*/
+            const regex = new RegExp(`(?=\\[(${tagNames})\\])`, 'gm');
+
+            /** iterating loop as long as the copy of the markdown text exists and split the copy as soon as a domainCommand identifier is found */
+            while (copy.length) {
+                /** As soon as an identifier is with regEx the copy of the markdown text is split and saved into {array} command
+                 *  split: saves its values into an {array}
+                 *  limit 1: indicated that as soon as an identifier is found copy is split */
+                const [command] = copy.split(regex, 1);
+                /** substring: reduces the {string} by the length in the brackets
+                 *  Split the copy by the length of {array} command to get the remaining array
+                 *  and save it into copy to start the loop again and search for further domainCommandIdentifiers
+                 *  when copy is empty the while loop will terminate*/
+                copy = copy.substring(command.length);
+                /** call the parseLineForDomainCommand for each extracted command
+                *   trim: reduced the whitespacing linebreaks */
+                const commandTextWithCommandIdentifier = this.parseLineForDomainCommand(command.trim());
+                /** push the commandTextWithCommandIdentifier into the commandTextsMappedToCommandIdentifiers*/
+                commandTextsMappedToCommandIdentifiers.push(commandTextWithCommandIdentifier);
+            }
+            /** emit the {array} commandTextsMappedToCommandIdentifiers to the client*/
+            this.textWithDomainCommandsFound.emit(commandTextsMappedToCommandIdentifiers);
+        }
+    }
+
+    /**
+     * @function parseLineForDomainCommand
+     * @desc Couple each textLine with the domainCommand identifier to emit that to the parent component for assignment
+     *       1. Check which domainCommand identifier is contained within the textLine
+     *       2. Remove the domainCommand identifier from the textLine
+     *       3. Create an array with first element textLine and second element the domainCommand identifier
+     * @param textLine {string} from the parse function
+     * @return array of the textLine with the domainCommand identifier
+     */
+    private parseLineForDomainCommand = (textLine: string): [string, DomainCommand] => {
+        for (const domainCommand of this.domainCommands) {
+            const possibleOpeningCommandIdentifier = [domainCommand.getOpeningIdentifier(), domainCommand.getOpeningIdentifier().toLowerCase(), domainCommand.getOpeningIdentifier().toUpperCase()];
+            // const possibleClosingIdentifier = [domainCommand.getClosingIdentifier(), domainCommand.getClosingIdentifier().toLowerCase(), domainCommand.getClosingIdentifier().toUpperCase()];
```

Comment:
If there is not a very good reason otherwise, commented out code should be removed as nobody will bother with it anyway.

Reply:
Updated.